Code/GraphMol/RGroupDecomposition/RGroupDecompData.h

```diff
@@ -435,8 +424,8 @@ struct RGroupDecompData {
     //  [m1_permutation_idx,  m2_permutation_idx, m3_permutation_idx]
 
     while (iterator.next()) {
-      if (count > N) {
-        throw ValueErrorException("Next did not finish");
+      if (count > iterator.maxPermutations) {
+        throw ValueErrorException("next() did not finish");
```

Comment:
note to myself, I always disliked this exception, it isn't meaningful to the user and really is a runtime condition.